pkg/pipelines/webhook/webhook.go

```diff
@@ -0,0 +1,223 @@
+package webhook
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/openshift/odo/pkg/pipelines/config"
+	"github.com/openshift/odo/pkg/pipelines/eventlisteners"
+	"github.com/openshift/odo/pkg/pipelines/git"
+	"github.com/openshift/odo/pkg/pipelines/ioutils"
+	"github.com/openshift/odo/pkg/pipelines/routes"
+	"github.com/openshift/odo/pkg/pipelines/secrets"
+)
+
+type webhookInfo struct {
+	clusterResource *resources
+	repository      *git.Repository
+	gitRepoURL      string
+	cicdNamepace    string
+	listenerURL     string
+	accessToken     string
+	names           []string
+	isCICD          bool
+}
+
+// Create creates a new webhook on the target Git Repository
+// names is a {envName, appName, seviceName} tuple.
+// It returns the ID of created webhook.
+func Create(accessToken, pipelinesFile string, names []string, isCICD bool) (string, error) {
+
+	webhook, err := newWebhookInfo(accessToken, pipelinesFile, names, isCICD)
+	if err != nil {
+		return "", err
+	}
+
+	exists, err := webhook.exists()
+	if err != nil {
+		return "", err
+	}
+
+	if exists {
+		return "", errors.New("webhook already exists")
+	}
+
+	return webhook.create()
+}
+
+// Delete deletes webhooks on the target Git Repository that match the listener address
+// names is a {envName, appName, seviceName} tuple.
+// It returns the IDs of deleted webhooks.
+func Delete(accessToken, pipelinesFile string, names []string, isCICD bool) ([]string, error) {
+
+	webhook, err := newWebhookInfo(accessToken, pipelinesFile, names, isCICD)
+	if err != nil {
+		return nil, err
+	}
+
+	ids, err := webhook.list()
+	if err != nil {
+		return nil, err
+	}
+
+	return webhook.delete(ids)
+}
+
+// List returns an array of webhook IDs for the target Git repository/listeners
+func List(accessToken, pipelinesFile string, names []string, isCICD bool) ([]string, error) {
+
+	webhook, err := newWebhookInfo(accessToken, pipelinesFile, names, isCICD)
+	if err != nil {
+		return nil, err
+	}
+
+	return webhook.list()
+}
+
+func newWebhookInfo(accessToken, pipelinesFile string, names []string, isCICD bool) (*webhookInfo, error) {
+
+	manifest, err := config.ParseFile(ioutils.NewFilesystem(), pipelinesFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse pipelines: %w", err)
+	}
+
+	if err := manifest.Validate(); err != nil {
+		return nil, err
+	}
+
+	gitRepoURL := getRepoURL(manifest, isCICD, names)
+	if gitRepoURL == "" {
+		return nil, errors.New("failed to find Git repostory URL in manifest")
+	}
+
+	cicdNamepace := getCICDNamespace(manifest)
```

Comment:
Don't you only need this if you're changing the CICD namespace?

i.e. it's not an error not to have a cicd namespace

Reply:
We need to know the CICD namespace between that is where the secrets and route are.